src/annotator/util/buckets.js

```diff
@@ -0,0 +1,248 @@
+import { getBoundingClientRect } from '../highlighter';
+
+/**
+ * @typedef {import('../../types/annotator').Anchor} Anchor
+ */
+
+/**
+ * A structured Array representing either the top (`startOrEnd` = 1) or the
```

Comment:
I think "tuple" is a common-enough term in JS circles these days for a structured array that you can use that.

Reply:
Done.